tests/src/Unit/AvPortalResourceTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\media_avportal\src\Unit;
+
+use Drupal\media_avportal\AvPortalResource;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests the AvPortalResource class.
+ */
+class AvPortalResourceTest extends UnitTestCase {
+
+  /**
+   * Tests the photo thumbnail preferred resolution.
+   */
+  public function testAvPortalPhotoThumbnailUrl(): void {
+    // Photo resource data array with all resolutions.
+    $data = [
+      'ref' => 'P-038924/00-15',
+      'doc_ref' => 'P-038924',
```

Comment:
Let's put only the minimum amount of data needed for this test. Let's remove the line `doc_ref`.